kimmobrunfeldt/git-hours
# Refactor codebase to modern (Airbnb) standards

## Refactor codebase

> Refactor codebase to modern (Airbnb) standards (project source code is nearly two years old)

### Changes

- Support Vim (ignore `*.swp` files)
- Specify Node engine `^12.22.3`
	- Older versions may work but provide little benefit
	- Version 14 takes a long time to compile Nodegit
	- Version 16 throws Nodegit compile errors
- Add Airbnb to eslint
	- Add new packages
		- eslint-config-airbnb-base
		- eslint-plugin-import
- Update packages
	- bluebird from `^2.1.3` to `^3.7.2`
	- commander from `^2.2.0` to `^8.0.0`
	- lodash from `^2.4.1` to `^4.17.21`
	- nodegit from `^0.26.2` to `^0.27.0`
	- eslint from `^1.5.1` to `^7.30.0`
	- mocha from `^2.3.3` to `^9.0.2`
	- releasor from `^0.1.3` to `^1.2.1`
- Remove packages
	- Remove jscs (is now in eslint)
- Fix 504 eslint errors
	-   1 `brace-style`
	-   1 `no-else-return`
	-   1 `no-plusplus`
	-   1 `no-redeclare`
	-   1 `object-shorthand`
	-   1 `padded-blocks`
	-   1 `prefer-destructuring`
	-   2 `block-scoped-var`
	-   2 `no-extra-parens`
	-   2 `space-after-keywords`
	-   2 `space-return-throw-case`
	-   3 `no-undef`
	-   4 `eqeqeq`
	-   5 `semi`
	-   9 `prefer-template`
	-  10 `operator-linebreak`
	-  15 `comma-dangle`
	-  20 `vars-on-top`
	-  37 `func-names`
	-  37 `prefer-arrow-callback`
	-  37 `space-before-function-paren`
	-  42 `no-var`
	- 345 `indent`
- Bump revision from `^1.4.0` to `^1.4.1`

### Semantics

- [ ] This PR will introduce breaking changes (major)
- [ ] This PR will introduce new functionality (minor)
- [x] This PR will fix or tweak existing functionality (patch)

### Categories

- [x] Configs
- [ ] Bugfixes
- [ ] File Structure
- [ ] Provisioning

### Validation Steps

1. Run `npm run dev` (runs `npm run test && npm run lint` on change)
1. Confirm no unit tests fail
1. Confirm no lint errors report
